### ISSUE TITLE
RFE: allow alias_target to be used with run_target

### ISSUE THREAD

**Issue by keszybz** [mesonbuild/meson#6186]
I'd like to aggregate a few run_targets() under a single command. Unfortunately this fails:

```
hwdb_update_ids = run_target('hwdb-update-ids', ....)
autosuspend_update = run_target('autosuspend-update', ...)
alias_target('hwdb-update', hwdb_update_ids, autosuspend_update)
```

hwdb.d/meson.build:56:0: ERROR: Depends items must be build targets.

**Comment by xclaesse** [comment]
The reason is run_target has no output file, so Ninja cannot use it as dependency. One trick is to use custom_target instead that write a dummy output file.

**Comment by keszybz** [comment]
I'd prefer if meson could handle this automatically for me. I don't know what the best way is, but even if a dummy file needs to be declared, then meson could do it internally.

**Comment by xclaesse** [comment]
I agree

**Comment by mattyclarkson** [comment]
This would be incredibly useful for linking multiple format/lint targets together.

**Comment by tp-m** [comment]
I'd also like to be able to use `alias_target()` with other `alias_target()s` - that should be possible, shouldn't it?

**Comment by xclaesse** [comment]
alias_target() is just a run_target() with no command to run internally, so it has the same limitation. I think both should be re-implemented internally with a wrapper script that creates a dummy output file so ninja can use that dummy file to know if the target has been built.

**Comment by jpakkane** [comment]
An alternative implementation would be that when the alias target is formed and it is given another alias target as a dependency, it will grab the dependencies from that target and add them to its own.

**Comment by xclaesse** [comment]
good point.

**Comment by eli-schwartz** [comment]
I uh, "rediscovered" this in #9175 (actually it started working during generation, and then created a broken build.ninja).

Also I then fixed it in #9202.

Anyway this works now. :)
